tracing-attributes/src/expand.rs

```diff
@@ -503,6 +507,22 @@ impl<'block> AsyncTraitInfo<'block> {
             }
         })?;
 
+        // is the last expression an async block?
+        if let Expr::Async(async_expr) = last_expr {
+            // check that the move 'keyword' is present
+            async_expr.capture?;
```

Comment:
hmm, should we try to handle functions returning `async` blocks _without_ the `move` keyword, too?